File: binary_search/Leetcode-349-IntersectionOfTwoArrays/Leetcode-349-IntersectionOfTwoArrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> s(nums1.begin(), nums1.end());
        vector<int> result;
        for (auto num : nums2) {
            if( s.count(num) ) {
                result.push_back(num);
                s.erase(num);
            }
        }
        return result;
    }
};

File: maureen72/BinarySearch/Leetcode-349-IntersectionofTwoArrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        result=[]
        set1=set(nums1)
        set2=set(nums2)
        result=(set1 & set2)
        return result
        